File: tests/cypress/cypress/elements/pages/utilities/translate.rb
class Translate < ControlPanelPage
  set_url_matcher /utilities\/translate/

  elements :languages, 'div.sidebar ul:nth-child(4) li'

  element :heading, 'div.w-12 form h1'

  element :phrase_search, 'form fieldset.tbl-search input[name=search]'
  element :search_submit_button, 'form fieldset.tbl-search input.submit'

  element :table, 'form table'
  element :no_results, 'form table tr.no-results'
  elements :rows, 'form table tr'

  element :bulk_action, 'form fieldset.tbl-bulk-act select[name="bulk_action"]'
  element :action_submit_button, 'form fieldset.tbl-bulk-act input.submit'

  def load
    self.open_dev_menu
    click_link 'Utilities'
    click_link 'English'
  end
end

File: tests/cypress/cypress/elements/pages/utilities/Translate.js
import ControlPanel from '../ControlPanel'

class Translate extends ControlPanel {
  constructor() {
    super()
    this.urlMatcher = /utilities\/translate/

    this.elements({
      'languages': 'div.sidebar ul:nth-child(4) li',

      'heading': '.ee-main .title-bar .title-bar__title',

      'phrase_search': 'form .search-input input[type!=hidden][name=search]',
      'search_submit_button': 'form fieldset.tbl-search input.submit',

      'table': 'form table',
      'no_results': 'form table tr.no-results',
      'rows': 'form table tr'
    })
  }

  load() {
    this.open_dev_menu()
    cy.get('body > .dropdown').find('a:contains("Utilities")').click()
    this.get('wrap').find('a:contains("CP Translation")').click()
    this.get('wrap').find('a:contains("English")').click()
  }

}
export default Translate;